versioned/persist/adapter/src/main/java/org/projectnessie/versioned/persist/adapter/spi/AbstractDatabaseAdapter.java

```diff
@@ -1047,6 +1075,15 @@ protected Hash copyCommits(
       } else {
         parents.add(0, targetHead);
       }
+      ByteString commitMetadata = resetWithMergeProps.apply(sourceCommit.getMetadata());
+      sourceCommit = CommitLogEntry.withNewMeta(sourceCommit, commitMetadata);
```

Comment:
Why this?

Reply:
Because of  - https://github.com/projectnessie/nessie/pull/2146#discussion_r719170422